packages/webapp/src/delegates/api/eden-contract.ts

```diff
@@ -18,14 +15,38 @@ const getMemberBudgetBalance = () => {
     return {}; // TODO
 };
 
-const getMemberElectionParticipationStatus = () => {
-    return {}; // TODO
-};
+const MEMBER_REPRESENTATIVE_IF_NOT_PARTICIPATED_IN_RECENT_ELECTION =
+    "zzzzzzzzzzzzj";
+// check that member has participated in an election (if there's been one yet) (!="zzz...") and came to consensus with their group in the last election (!=0)
+const memberHasRepresentative = (member: EdenMember) =>
+    member.account !== "" &&
+    member.account !==
+        MEMBER_REPRESENTATIVE_IF_NOT_PARTICIPATED_IN_RECENT_ELECTION;
 
-const hasMemberRSVPed = () => {
-    return false; // TODO
-};
+export const getMyDelegation = async (
+    loggedInMemberAccount: string | undefined
+): Promise<EdenMember[]> => {
+    let myDelegates: EdenMember[] = [];
 
-export const getMyDelegation = () => {
-    return {}; // TODO
+    if (!loggedInMemberAccount) return myDelegates;
+
+    const leadRepresentative = await getHeadDelegate();
+    let nextDelegate = await getEdenMember(loggedInMemberAccount);
```

Comment:
We should avoid making API network calls directly outside of `react-query`. We lose a lot of benefits (caching, speed, reductions in network traffic, etc.)

You can call these imperatively here using `queryCache.fetchQuery()` or create a custom hook as in this example: https://github.com/eoscommunity/Eden/pull/358.